clash-cores/src/Clash/Cores/Xilinx/DcFifo.hs

```diff
@@ -0,0 +1,271 @@
+{-|
+Copyright  :  (C) 2022, Google Inc,
+License    :  BSD2 (see the file LICENSE)
+Maintainer :  QBayLogic B.V. <devops@qbaylogic.com>
+
+Support for the [FIFO Generator v13.2](https://docs.xilinx.com/v/u/en-US/pg057-fifo-generator).
+
+It is necessary to read the product guide linked above in order to effectively
+use the FIFO. The product guide also documents how to interface the FIFO
+correctly. To aid comprehension, the signals in this module also mention the
+name the product guide uses for that signal like @this@.
+
+Note that the behavior of the FIFO in Haskell simulation does not correspond
+exactly to the behavior in HDL simulation, and that neither behaviors correspond
+exactly to hardware. All the different behaviors are functionally correct and
+equivalent when the usage guidelines in the product guide are observed.
+
+Furthermore, the FIFO is configured as follows:
+
+* /Native/ interface type
+* /Independent clocks block RAM/ implementation
+* 2 synchronization stages
+* /Standard FIFO/ read mode
+* /No/ output registers
+* Reset pin /enabled/, reset synchronization /disabled/, safety circuit
+/disabled/
+
+    (Note: the GUI will still say /Asynchronous Reset/, grayed out. This might be
+    confusing: the resets are actually synchronous.)
+
+* Full flag reset value: /@0 (False)@/
+
+    (Note: this is actually not configurable for synchronous resets, it is
+    always @False@.)
+
+* No /Dout reset value/
+* Optional overflow flag: 'dcOverflow'
+* Optional underflow flag: 'dcUnderflow'
+* No other configurable status flags (no almost full, almost empty, write
+acknowledge, valid, or programmable full\/empty flags)
+* Optional write data count: 'dcWriteDataCount' (only full width is supported)
+* Optional read data count: 'dcReadDataCount' (only full width is supported)
+
+(The order of these items corresponds to Vivado's /Customize IP/ GUI dialog in
+Vivado 2022.1.)
+
+== Reset sequencing
+
+The FIFO only supports synchronous resets ('Clash.Signal.Synchronous'). If the
+read or write domain has an asynchronous reset signal, simulating the FIFO or
+generating HDL for it will throw an error.
+
+The FIFO does not need to be reset; it is possible to use the FIFO with its
+reset inputs permanently deasserted. However, resetting the FIFO will flush it,
+so resets do have a use. It is necessary to generate a proper reset sequence;
+see the product guide for details. Read and write side need to be reset
+together, both for at least one clock cycle. During the time where either or
+both of the resets are asserted, do not perform write or read operations to
+avoid unexpected behavior.
+
+-}
+
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE ViewPatterns #-}
+
+{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
```

Comment:
I don't think we need this anymore. I think this was here to suppress an HLint pragma.

Reply:
Oh true.